### PR TITLE
compat with sympy 1.2 and requirements

### LINKED ISSUE #615: sympy 1.12 currently breaks devito

### PR DESCRIPTION
**Pull request by mloubout**
Update custom `Indexed` and `.dot` deprecation for symyp 1.2 

fixes #615

### CODE REVIEW

**Comment by mloubout** [comment]
Freaking new version of `pip` is screwing stuff up

File: requirements.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 numpy==1.14
-sympy==1.1
```
Comment (FabioLuporini) [inline requirements.txt]: the ``==`` was actually a hack given that 1.12 wasn't working. Can we rather have ``>= 1.1``? or does turning ``dot`` into ``*`` prevent this? I see travis is still having, so you can easily check this with another round of CI

Comment (mloubout) [reply requirements.txt]: Not sure ,lemme try

Comment (mloubout) [reply requirements.txt]: Looks like is fine with `>=`

**Comment by FabioLuporini** [comment]
uncontroversial. Thanks for this, Mathias.

**Pull request merged**